tests/test_flash_attn.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+
+import pytest
+import torch
+
+from llmfoundry.models.layers.attention import (flash_attn_fn,
+                                                is_flash_v2_installed,
+                                                triton_flash_attn_fn)
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize('kv_n_heads', [1, 4, 8])
+def test_gqa_kv_repetition(kv_n_heads: int):
+    # Test that flash attention v2 with GQA (kv_n_heads < n_heads) works the same
+    # whether we repeat the kv_n_heads explicitly or flash attention v2 handles it on its own.
+    if not is_flash_v2_installed():
```

Comment:
move this to `@pytest.skipif(<condition>)`

Reply:
Done.